tools/autograph-client/client.go

```diff
@@ -282,16 +313,31 @@ func verifyContentSignature(input []byte, resp signatureresponse, endpoint strin
 	return ecdsa.Verify(pubKey, input, sig.R, sig.S)
 }
 
-func verifyXPI(input []byte, resp signatureresponse) bool {
-	sig, err := xpi.Unmarshal(resp.Signature, input)
-	if err != nil {
-		log.Fatal(err)
-	}
-	err = sig.VerifyWithChain(nil)
-	if err != nil {
-		log.Fatal(err)
+func verifyXPI(input []byte, req signaturerequest, resp signatureresponse, reqType requestType) bool {
```

Comment:
`req signaturerequest` already contains the request URI so you don't really need to pass `reqType`

Reply:
How do I access that? Is it in Options?

```go
type signaturerequest struct {
        Input   string `json:"input"`
        KeyID   string `json:"keyid"`
        Options interface{}
}
```

Reply:
fail. it's not there, I misread `signaturerequest` for `http.Request`. nevermind.